internal/mtls/mtls.go

```diff
@@ -0,0 +1,130 @@
+package mtls
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc/credentials"
+	exp_credentials "google.golang.org/grpc/experimental/credentials"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	hubbleRelayExpectedServerName = "ui.hubble-relay.cilium.io"
+)
+
+var (
+	ErrCertDataMissingInSecret = errors.New("required certificate data (ca.crt, tls.crt, or tls.key) not found in secret")
+)
+
+// getHubbleMTLSCertificatesFromSecret fetches ca.crt, tls.crt (client cert), and tls.key (client key)
+// from the specified Kubernetes secret.
+func getHubbleMTLSCertificatesFromSecret(
+	ctx context.Context,
+	clientset kubernetes.Interface,
+	logger *zap.Logger,
+	secretName string,
+	secretNamespace string,
+) (caData, clientCertData, clientKeyData []byte, err error) {
+
+	logger.Info("Fetching mTLS certificates from Kubernetes secret",
+		zap.String("secretName", secretName),
+		zap.String("secretNamespace", secretNamespace))
```

Comment:
1. Info is too verbose for this, IMHO.
1. Consistently use camel_case for field names like for the rest of the codebase.
```suggestion
	logger.Debug("Fetching mTLS certificates from Kubernetes secret",
		zap.String("name", secretName),
		zap.String("namespace", secretNamespace))
```